### PR TITLE
Added ability to access request headers from proxytest

### PR DESCRIPTION
**Pull request by Zagidziran**
Added ability to retrieve request headers from proxytest

### CODE REVIEW

**Comment by mathetake** [comment]
Could you just add SetHeader call in in `http_headers` example, rather than adding a new example just for setHeader? https://github.com/tetratelabs/proxy-wasm-go-sdk/tree/main/examples/http_headers

File: proxytest/http.go [hunk 1]
```diff
@@ -405,6 +405,16 @@ func (h *httpHostEmulator) GetCurrentHttpStreamAction(contextID uint32) types.Ac
 	return stream.action
 }
 
+// impl HostEmulator
+func (h *httpHostEmulator) GetHttpRequestHeaders(contextID uint32) types.Headers {
+	stream, ok := h.httpStreams[contextID]
+	if !ok {
+		log.Fatalf("invalid context id: %d", contextID)
+	}
+
```
Comment (mathetake) [inline proxytest/http.go:414]: delete the empty line so as to be aligned with `GetCurrentHttpStreamAction`

File: proxytest/proxytest.go [hunk 2]
```diff
@@ -58,6 +58,7 @@ type HostEmulator interface {
 	CallOnRequestBody(contextID uint32, body []byte, endOfStream bool) types.Action
 	CompleteHttpContext(contextID uint32)
 	GetCurrentHttpStreamAction(contextID uint32) types.Action
+	GetHttpRequestHeaders(contextID uint32) types.Headers
```
Comment (mathetake) [inline proxytest/proxytest.go:61]: please rename to `GetRequestHeaders` to be better aligned.

Comment (Zagidziran) [reply proxytest/proxytest.go:61]: Can you provide your option?

Comment (mathetake) [reply proxytest/proxytest.go:61]: I mean make this aligned with `GetCurrentHttpStreamAction` right above which has same functionality for streamAction.

File: examples/http_headers/main_test.go [hunk 3]
```diff
@@ -31,6 +31,20 @@ func TestHttpHeaders_OnHttpRequestHeaders(t *testing.T) {
 	// Call OnHttpStreamDone.
 	host.CompleteHttpContext(id)
 
+	// Check headers.
+	resultHeaders := host.GetCurrentRequestHeaders(id)
+
```
Comment (mathetake) [inline examples/http_headers/main_test.go:36]: delete unnecessary empty line.

File: examples/http_headers/main_test.go [hunk 4]
```diff
@@ -31,6 +31,20 @@ func TestHttpHeaders_OnHttpRequestHeaders(t *testing.T) {
...
+	// Check headers.
+	resultHeaders := host.GetCurrentRequestHeaders(id)
+	
+	var headerValue *string
+	
+	for _, val := range resultHeaders {
+        if val[0] == "test" {
+            headerValue = &val[1]
+		}
+    }
```
Comment (mathetake) [inline examples/http_headers/main_test.go:43]: please run gofmt.

File: examples/http_headers/main_test.go [hunk 5]
```diff
@@ -31,6 +31,20 @@ func TestHttpHeaders_OnHttpRequestHeaders(t *testing.T) {
...
+	
+	require.NotNil(t, headerValue)
+    assert.Equal(t, *headerValue, "best")
```
Comment (mathetake) [inline examples/http_headers/main_test.go:46]: ditto.

Comment (mathetake) [inline examples/http_headers/main_test.go:46]:
```suggestion
	var found bool
	for _, val := range resultHeaders {
		if val[0] == "test" {
			require.Equal(t, val[1], "best")
			found = true
		}
	}
	require.True(t, found)
```

**Review by mathetake — APPROVED** [review]

File: examples/http_headers/main_test.go [hunk 6]
```diff
@@ -31,6 +31,17 @@ func TestHttpHeaders_OnHttpRequestHeaders(t *testing.T) {
...
+	resultHeaders := host.GetCurrentRequestHeaders(id)
+	var found bool
+	for _, val := range resultHeaders {
+		if val[0] == "test" {
+			require.Equal(t, "best", val[1])
+			found = true
+		}
+	}
+	require.True(t, found)
+
```
Comment (mathetake) [inline examples/http_headers/main_test.go:44]: one last thing: can you move this before `host.CompleteHttpContext(id)` above?

**Pull request merged**